### ISSUE TITLE
Not working with Safari 10

### ISSUE THREAD

**Issue by mfkp** [amw/jpeg_camera#14]
This is more of an issue with swfobject, but since that's an unmaintained project, might be worth finding a fix here.

Starting in Safari 10, Flash is "click-to-enable", so when `hasFlashPlayerVersion` is called, it returns false, and doesn't attempt to load the flash fallback:

https://github.com/amw/jpeg_camera/blob/master/src/jpeg_camera_flash.coffee#L11

I'm trying to think of a good way to workaround this. A simple useragent detection could probably do the trick, but seems hacky. Any other ideas?

**Comment by amw** [comment]
Thanks for bringing this up. Yeah, it's a tricky one.

We could simply skip flash detection and always show the SWF object when our HTML5 features are unavailable. The only situation where it doesn't make sense is mobile where you can't get flash anyway. But having empty flash space on iPad is not as nice as my customized message apologizing for my application not being able to run.

I could change the API and report progress of engine initialization. Propagate the swfobject [success/failure callback](https://github.com/amw/jpeg_camera/blob/master/src/jpeg_camera_flash.coffee#L62) back to the API user. The user could wait for that, control timeout on their own and communicate with the user accordingly.

I'll start with simply always falling back to flash when HTML5 is not available and I may add the smarts about progress reporting later.

**Comment by mfkp** [comment]
Yeah, that should be enough to get it working on desktop Safari again (after they click to enable it). Let me know if you need any help testing it out.

Hopefully Apple will add GetUserMedia to Safari sometime within the next 10 years.

**Comment by amw** [comment]
I have tried simply skipping `can_use_flash` and always falling back on flash, but that's not a good solution. When there is no flash installed 1) developer doesn't know about it, 2) user doesn't see "unsupported" message.

A more sophisticated solution would be required, but I don't have time or incentives to develop it at this time. I will accept a PR if someone suggests a reasonable, clean approach.

**Comment by freddi301** [comment]
add this fallback

```html
<input type="file" accept="image/*" capture="camera">
<input type="file" accept="video/*" capture="camera">
```
